### PR TITLE
Gallery

### PR DESCRIPTION
**Pull request by anxolerd**
Refactored `gallery.js`

## Changelog
- smoother work on IE
- supports swipes (TODO: test this better)

### CODE REVIEW

File: project/static/src/js/gallery.js [hunk 1]
```diff
@@ -1,84 +1,97 @@
...
-        if (isNaN(fraction)) { return; }
+  var DIRECTIONS = {
+    LEFT: -1,
+    NONE: 0,
+    RIGHT: 1,
+  }
 
-        el.scrollLeft = startPosition + delta * fraction;
+  $.fn.doScroll = function(duration, direction) {
+    var $elem = this;
```
Comment (Cybran111) [inline project/static/src/js/gallery.js]: Use
function(...){...}.bind(this) if there is a problem with a context.

File: project/static/src/js/gallery.js [hunk 2]
```diff
@@ -1,84 +1,97 @@
...
-        mainClass: 'mfp-with-zoom mfp-img-mobile',
-        image: {
-            verticalFit: true
-        },
-        gallery: {
-            enabled: true,
-            preload: [1, 1]
-        }
+  document.addEventListener("DOMContentLoaded", (function () {
+    require("magnific-popup");
```
Comment (Cybran111) [inline project/static/src/js/gallery.js]: Using require() in the middle of code is an anti-pattern

**Comment by Cybran111** [comment]
What about freezes in browser that have been reported in #162?

**Comment by anxolerd** [comment]
They are gone in IE
However they appeared when you try both swiping and moving mouse to the edge

File: project/static/src/js/gallery.js [hunk 3]
```diff
@@ -1,84 +1,98 @@
-var $ = require('npm-zepto');
+(function ($) {
```
Comment (Orhideous) [inline project/static/src/js/gallery.js]: Вместо
```js
(function ($) {
    $.fn.badPlugin = function(){}
})(require("npm-zepto"));
```
Надо
```js
(function (factory) {
    if (typeof define === 'function' && define.amd) {
        // AMD. Register as an anonymous module.
        define(['npm-zepto'], factory);
    } else if (typeof module === 'object' && module.exports) {
        // Node/CommonJS
        module.exports = factory(require('npm-zepto'));
    } else {
        // Browser globals
        factory($);
    }
}(function ($) {
    $.fn.jqueryPluginCommonJs = function () {};
}));
```
https://raw.githubusercontent.com/umdjs/umd/master/jqueryPluginCommonjs.js

File: project/static/src/js/gallery.js [hunk 4]
```diff
@@ -1,84 +1,98 @@
...
-        closeBtnInside: false,
-        mainClass: 'mfp-with-zoom mfp-img-mobile',
-        image: {
-            verticalFit: true
-        },
-        gallery: {
-            enabled: true,
-            preload: [1, 1]
-        }
+  document.addEventListener("DOMContentLoaded", (function () {
```
Comment (Orhideous) [inline project/static/src/js/gallery.js]: > addEventListener

И у zeptojs, и у jQuery есть метод `.on()`

http://zeptojs.com/#on

**Pull request merged**